### PR TITLE
Core/LootMgr: Update lootMgr to normalize loot_template tables

### PR DESCRIPTION
**Pull request by Viste**
CHANGES PROPOSED:  Update the structure of *_loot_templates
Closes: #1120
Target branch(es): master

### CODE REVIEW

**Comment by Winfidonarleyan** [comment]
Good work

File: src/server/game/Loot/LootMgr.h [hunk 1]
```diff
@@ -115,19 +115,20 @@ struct Loot;
...
     ConditionList conditions;                               // additional loot condition
 
-    // Constructor, converting ChanceOrQuestChance -> (chance, needs_quest)
+    // Constructor
     // displayid is filled in IsValid() which must be called after
-    LootStoreItem(uint32 _itemid, float _chanceOrQuestChance, uint16 _lootmode, uint8 _group, int32 _mincountOrRef, uint8 _maxcount)
-        : itemid(_itemid), chance(fabs(_chanceOrQuestChance)), mincountOrRef(_mincountOrRef), lootmode(_lootmode),
-        group(_group), needs_quest(_chanceOrQuestChance < 0), maxcount(_maxcount)
+    LootStoreItem(uint32 _itemid, uint32 _reference, float _chance, bool _needs_quest, uint16 _lootmode, uint8 _groupid, int32 _mincount, uint8 _maxcount)
+        : itemid(_itemid), reference(_reference), chance(_chance), lootmode(_lootmode),
```
Comment (Nefertumm) [inline src/server/game/Loot/LootMgr.h]: Please, re-order the initialization. Put loot_mode after needs_quest to fix travis ;)
```suggestion
        : itemid(_itemid), reference(_reference), chance(_chance), needs_quest(_needs_quest), lootmode(_lootmode),
```

**Comment by Viste** [comment]
thx Winfidonarleyan for DB fixes

File: src/server/game/Loot/LootMgr.cpp [hunk 2]
```diff
@@ -301,53 +308,53 @@ bool LootStoreItem::Roll(bool rate, Player const *player, Loot& loot, LootStore
...
-    if (group >= 1 << 7)                                     // it stored in 7 bit field
+    if (groupid >= 1 << 7)                                     // it stored in 7 bit field
     {
-        sLog->outErrorDb("Table '%s' entry %d item %d: group (%u) must be less %u - skipped", store.GetName(), entry, itemid, group, 1 << 7);
+        sLog->outErrorDb("Table '%s' Entry %d Item %d: GroupId (%u) must be less %u - skipped", store.GetName(), entry, itemid, groupid, 1 << 7);
         return false;
     }
 
-    if (mincountOrRef == 0)
+    if (reference == 0)
```
Comment (Nefertumm) [inline src/server/game/Loot/LootMgr.cpp]:
```suggestion
    if (mincount== 0)
```

File: src/server/game/Loot/LootMgr.cpp [hunk 3]
```diff
@@ -301,53 +308,53 @@ bool LootStoreItem::Roll(bool rate, Player const *player, Loot& loot, LootStore
...
     {
         sLog->outErrorDb("Table '%s' entry %d item %d: wrong mincountOrRef (%d) - skipped", store.GetName(), entry, itemid, mincountOrRef);
```
Comment (Nefertumm) [inline src/server/game/Loot/LootMgr.cpp]:
```suggestion
         sLog->outErrorDb("Table '%s' Entry %d Item %d: wrong MinCount (%d) - skipped", store.GetName(), entry, itemid, mincount);
```

File: src/server/game/Loot/LootMgr.cpp [hunk 4]
```diff
@@ -301,53 +308,53 @@ bool LootStoreItem::Roll(bool rate, Player const *player, Loot& loot, LootStore
...
         return false;
     }
 
-    if (mincountOrRef > 0)                                  // item (quest or non-quest) entry, maybe grouped
+    if (reference > 0)                                  // item (quest or non-quest) entry, maybe grouped
```
Comment (Nefertumm) [inline src/server/game/Loot/LootMgr.cpp]:
```suggestion
    if (reference == 0)                                  // item (quest or non-quest) entry, maybe grouped
```

File: src/server/game/Loot/LootMgr.cpp [hunk 5]
```diff
@@ -301,53 +308,53 @@ bool LootStoreItem::Roll(bool rate, Player const *player, Loot& loot, LootStore
...
-            sLog->outErrorDb("Table '%s' entry %d item %d: low chance (%f) - skipped",
+            sLog->outErrorDb("Table '%s' Entry %d Item %d: low chance (%f) - skipped",
                 store.GetName(), entry, itemid, chance);
             return false;
         }
 
-        if (maxcount < mincountOrRef)                       // wrong max count
+        if (maxcount < mincount)                       // wrong max count
         {
             sLog->outErrorDb("Table '%s' entry %d item %d: max count (%u) less that min count (%i) - skipped", store.GetName(), entry, itemid, int32(maxcount), mincountOrRef);
```
Comment (Nefertumm) [inline src/server/game/Loot/LootMgr.cpp]:
```suggestion
             sLog->outErrorDb("Table '%s' Entry %d Item %d: MaxCount (%u) less that MinCount (%i) - skipped", store.GetName(), entry, itemid, int32(maxcount), mincount);
```

File: src/server/game/Loot/LootMgr.cpp (hunk 4 above)
Comment (Viste) [reply src/server/game/Loot/LootMgr.cpp]: no here ref must be > 0 ref == 0 is early

Comment (Nefertumm) [reply src/server/game/Loot/LootMgr.cpp]: Changed according to this line https://github.com/TrinityCore/TrinityCore/commit/55b43c67519359f0e5a96004c393898b3c62add3#diff-469201924aa4feec87dd70a2bf60bb14R310

**Comment by wetbrownsauce** [comment]
I'm running this PR on my server, not seeing any conflicts or issues so far, but I haven't had much of a chance to test due to the holidays. I'll keep you posted!

**Comment by FrancescoBorzi** [comment]
@Viste @wetbrownsauce thanks a lot. This can now be merged.

**Comment by FrancescoBorzi** [comment]
@Viste do you want more feedback?

**Pull request merged**